src/services/healthcareProfessionalService.ts

```diff
@@ -0,0 +1,1033 @@
+ * - Can update scalar fields (names, degrees, specialties, etc.)
+ * - Can update facility relationship (but must keep at least one)
+ * - All changes are atomic (scalar + relations + audit log)
+ * @param id - HP ID to update
+ * @param fieldsToUpdate - Partial update input from GraphQL
+ * @param updatedBy - User performing the update (for audit)
+ * @returns Result with updated HealthcareProfessional or errors
+ */
+export const updateHealthcareProfessional = async (
+    id: string,
+    fieldsToUpdate: Partial<gqlTypes.UpdateHealthcareProfessionalInput>,
+    updatedBy: string
+): Promise<Result<gqlTypes.HealthcareProfessional>> => {
+    try {
+        // Validate input before starting transaction
+        const validationResult = validateUpdateProfessionalInput(fieldsToUpdate)
+
+        if (validationResult.hasErrors) {
+            return validationResult as Result<gqlTypes.HealthcareProfessional>
+        }
+
+        // Execute all operations in atomic transaction
+        const gqlHealthcareProfessional = await db.transaction().execute(async trx => {
+            // Fetch current HP state
+            const currentHp = await trx
+                .selectFrom('hps')
+                .selectAll()
+                .where('id', '=', id)
+                .executeTakeFirst()
+
+            if (!currentHp) {
+                throw new Error('HP_NOT_FOUND')
+            }
+
+            // Fetch current facility relations
+            const currentRelations = await trx
+                .selectFrom('hps_facilities')
+                .select('facilities_id')
+                .where('hps_id', '=', id)
+                .execute()
+
+            const currentFacilityIds = currentRelations.map(r => r.facilities_id)
+
+            // Build update patch for scalar fields
+            const updatePatch = buildHpUpdatePatch(fieldsToUpdate)
+
+            // Update scalar fields (always touch updatedDate)
+            const hasScalarChanges = Object.keys(updatePatch).length > 1 // More than just updatedDate
+
+            let updatedHp = currentHp
+
+            if (hasScalarChanges) {
+                updatedHp = await trx
+                    .updateTable('hps')
+                    .set(updatePatch)
+                    .where('id', '=', id)
+                    .returningAll()
+                    .executeTakeFirstOrThrow()
+            } else {
+                // Just touch updatedDate to track the change
+                updatedHp = await trx
+                    .updateTable('hps')
+                    .set({ updatedDate: new Date().toISOString() })
+                    .where('id', '=', id)
+                    .returningAll()
+                    .executeTakeFirstOrThrow()
+            }
+
+            // Handle facility relationship changes (if provided)
+            let finalFacilityIds = currentFacilityIds
+
+            if (fieldsToUpdate.facilityIds !== undefined) {
+                // Resolve new facility ID from relationship actions
```

Comment:
let's break down this long method into some smaller ones. 
This for example, in the old file was something like updateFacilityRelationships(). Let's do something similar here